packages/driver/src/cypress/location.ts

```diff
@@ -16,6 +16,23 @@ const reFile = /^file:\/\//
 const reLocalHost = /^(localhost|0\.0\.0\.0|127\.0\.0\.1)/
 const reQueryParam = /\?[^/]+/
 
+export interface LocationObject {
```

Comment:
Can we move to `driver/types/cypress/location.d.ts`?

Reply:
Any reason to do so? My understanding is that it's better to keep the types close to where they're relevant and that we rely on `.d.ts` files because a lot of the driver isn't properly typed.

Reply:
I moved my types this direction because @flotwig had asked me to here so I assumed it was the standard we were moving forward with:
https://github.com/cypress-io/cypress/pull/20474#discussion_r822826899

If this isn't the standard we want to go forward with, I'm fine with that (I'd prefer non-globals) but want to be aligned as a team on the correct approach.

Reply:
`internal-types.d.ts` is good for when we want to extend the `Cypress` namespace with our own types. Otherwise, it's ok to export types from `.ts`.

Reply:
What do you mean exactly?

Reply:
Basically what it says at the top of the file:

```
// NOTE: this is for internal Cypress types that we don't want exposed in the public API but want for development
```

If you look at the types defined in `internal-types.d.ts`, it's all extending the existing `Cypress` namespace exported by the `cli/types/cypress.d.ts`. This way, we can access private members of the public API in a type-safe way.